BrainPortal/lib/boutiques_save_std_out_std_err.rb

```diff
@@ -0,0 +1,151 @@
+#     "stdout_output_dir": "",
+#     "stderr_output_dir": "path/to/dir"
+#   }
+# }
+#
+# In case of a MultilevelSshDataProvider the "path/to/dir" will be use to save the output.
+# In case of a no MultilevelSshDataProvider the "path/to/dir" will be ignored.
+#
+# The value of the key "stdout_output_dir" and "stderr_output_dir" can be set to an empty string,
+# in this situation the files will be saved directly in the root folder of the DataProvider.
+module BoutiquesSaveStdOutStdErr
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  # This method overrides the one in BoutiquesClusterTask.
+  # Save the stdout and stderr files of the task as output files.
+  # The files will be saved as a child of the first input file.
+  def save_results
+    # Get the folder where to save the log files from the descriptor
+    descriptor      = self.descriptor_for_save_results
+    module_info = descriptor.custom_module_info('BoutiquesSaveStdOutStdErr')
+
+    # Get parent file to set stderr and stdout as children of first input file
+    main_input_id = descriptor.file_inputs.first.id
+    file_id       = self.invoke_params[main_input_id]
+    parent_file   = Userfile.find(file_id)
+
+    # Save stdout
+    science_stdout_basename = science_stdout_basename(self.run_number)
+    save_stdout_basename    = (Pathname.new(module_info["stdout_output_dir"]) +
+                              "#{self.pretty_type}-#{self.bname_tid_dashed}").to_s
+    stdout_file             = save_log_file(science_stdout_basename, save_stdout_basename, parent_file)
+    self.params["_cbrain_output_cbrain_stdout"] = [stdout_file.id] if stdout_file
+
+    # Save stderr
+    science_stderr_basename = science_stderr_basename(self.run_number)
+    save_stderr_basename    = (Pathname.new(module_info["stderr_output_dir"]) +
+                              "#{self.pretty_type}-#{self.bname_tid_dashed}").to_s
+    stderr_file             = save_log_file(science_stderr_basename, save_stderr_basename, parent_file)
+    self.params["_cbrain_output_cbrain_stderr"] = [stderr_file.id] if stderr_file
+
+    self.save
+
+    super
+  end
+
+  # Add the stdout and stderr files to the descriptor
+  # for the show page of the task.
+  def descriptor_for_show_params  #:nodoc:
+    descriptor = super.dup
+
+    stdout_file = BoutiquesSupport::OutputFile.new({
+      "id"   => "cbrain_stdout",
+      "name" => "stdout",
+      "description" => "Standard output of the tool",
+      "optional" => true
+    })
+
+    stderr_file = BoutiquesSupport::OutputFile.new({
+      "id"   => "cbrain_stderr",
+      "name" => "stderr",
+      "description" => "Standard error of the tool",
+      "optional" => true
+    })
+
+    descriptor["output-files"] << stdout_file if !descriptor["output-files"].find { |f| f["id"] == "stdout" }
```

Comment:
The comparison with the ID is wrong, because the ID is `cbrain_stdout` not `stdout`